cmd/bcpd/app/app_test.go

```diff
@@ -646,19 +627,32 @@ func benchmarkSendTxWithMultisig(b *testing.B, nbAccounts, blockSize, nbContract
 		c.id = createContract(b, myApp, chainID, int64(i+1), []*account{signer}, threshold, c.sigs()...)
 	}
 
-	txs := make([][]byte, b.N*blockSize)
+	txs := make([][]byte, b.N)
 	for i := 0; i < b.N; i++ {
-		for k := 0; k < blockSize; k++ {
-			sender := contracts[cmn.RandInt()%nbContracts]
-			recipient := contracts[cmn.RandInt()%nbContracts]
-			txs[(i*blockSize)+k] = makeSendTxMultisig(b, chainID, sender, recipient, "ETH", "benchmark", 1)
-		}
+		sender := contracts[cmn.RandInt()%nbContracts]
+		recipient := contracts[cmn.RandInt()%nbContracts]
+		txs[i] = makeSendTxMultisig(b, chainID, sender, recipient, "ETH", "benchmark", 1)
 	}
 
 	b.ResetTimer()
 
-	for i := 0; i < b.N; i++ {
-		newBlock(b, myApp, txs, nbAccounts, blockSize, i, nbContracts+1)
+	for i := 1; i <= b.N; i++ {
+		chres := myApp.CheckTx(txs[i-1])
+		require.Equal(b, uint32(0), chres.Code, chres.Log)
+
+		if i%blockSize == 0 {
+			header := abci.Header{Height: int64(i/blockSize) + 1}
+			myApp.BeginBlock(abci.RequestBeginBlock{Header: header})
+
+			for k := i - blockSize; k < i; k++ {
```

Comment:
okay, this confused me at first, until I realized starting at one means this if loop doesn't get hit until i:=blockSize

this is good.

I would just add one more thing at the end, cuz if b.N is not a multiple of blockSize, then the last chunck will never make a block.

eg. b.N = 500, blockSize = 200... last 100 tx would never be delivered.